src/dotnet/ToolManifest/ToolManifestFinder.cs

```diff
@@ -35,8 +35,39 @@ public IReadOnlyCollection<ToolManifestPackage> Find(FilePath? filePath = null)
                     ? new[] {(filePath.Value, filePath.Value.GetDirectoryPath())}
                     : EnumerateDefaultAllPossibleManifests();
 
+            var findAnyManifest =
+                TryFindToolManifestPackages(allPossibleManifests, out var toolManifestPackageAndSource);
+
+            if (!findAnyManifest)
+            {
+                throw new ToolManifestCannotBeFoundException(
+                    string.Format(LocalizableStrings.CannotFindAnyManifestsFileSearched,
+                        string.Join(Environment.NewLine, allPossibleManifests.Select(f => f.manifestfile.Value))));
+            }
+
+            return toolManifestPackageAndSource.Select(t => t.toolManifestPackage).ToArray();
+        }
+
+        public IReadOnlyCollection<(ToolManifestPackage toolManifestPackage, FilePath SourceManifest)> Inspect(
+            FilePath? filePath = null)
+        {
+            IEnumerable<(FilePath manifestfile, DirectoryPath _)> allPossibleManifests =
+                filePath != null
+                    ? new[] {(filePath.Value, filePath.Value.GetDirectoryPath())}
+                    : EnumerateDefaultAllPossibleManifests();
+
+            TryFindToolManifestPackages(allPossibleManifests, out var toolManifestPackageAndSource);
```

Comment:
Should you have some condition here for the case where this fails? Given that it is a Try... method?

Reply:
fixed, added false case. it happens to work since toolManifestPackageAndSource will not be null but empty array when returns false.